llvm/test/MC/RISCV/xqcibi-relocations.s

```diff
@@ -19,6 +21,15 @@ qc.e.bgeui x8, 12, foo
 # RELOC: R_RISCV_JAL foo 0x0
 # INSTR: qc.e.bgeui s0, 12, foo
 
+# Check that we print the correct relocations in exact mode
+.option exact
+qc.e.bgeui x8, 12, foo
+# RELOC: R_RISCV_VENDOR QUALCOMM 0x0
```

Comment:
For relocation generation testing we should use llvm-objdump -dr to test r_offset and test that we don't incorrectly emit redundant relocations. See Sparc/Relocations/relocation-specifier.s for a modern example using llvm-objdump -dr

Looks like R_RISCV_VENDOR is only generated with `exact`. We should test that without `exact` there is no `QUALCOMM`. This is unclear from the PR description.

Reply:
QUALCOMM should be emitted even without `exact`, we have started to use exact to avoid instruction compression happening and make testing more specific.

I will move to `objdump -dr`, I agree it's often clearer. I will revisit these tests.